extra_geom/detectors.py

```diff
@@ -854,6 +856,159 @@ def to_distortion_array(self, allow_negative_xy=False):
         return super().to_distortion_array(allow_negative_xy)
 
 
+class AGIPD_500K2GGeometry(DetectorGeometryBase):
+    """Detector layout for AGIPD-500k
+
+    The coordinates used in this class are 3D (x, y, z), and represent metres.
+
+    You won't normally instantiate this class directly:
+    use one of the constructor class methods to create or load a geometry.
+    """
+    detector_type_name = 'AGIPD-500K2G'
+    pixel_size = 2e-4  # 2e-4 metres == 0.2 mm
+    frag_ss_pixels = 64
+    frag_fs_pixels = 128
+    expected_data_shape = (8, 512, 128)
+    n_modules = 8
+    n_tiles_per_module = 8
+
+    @classmethod
+    def from_center(cls, center, asic_gap=2, panel_gap=(16, 30), unit=pixel_size):
+        """Generate an AGIPD-500K2G geometry from center position.
+
+        This produces an idealised geometry, assuming all modules are perfectly
+        flat, aligned and equally spaced within their quadrant.
+
+        The center position is given in pixel units, referring to the first
+        pixel of the first module, corresponding to data channels 0.
+
+        The origin of the coordinates is the bottom-right corner of the
+        first module of the detector (channel 0). Coordinates increase upwards
+        and to the left (looking along the beam).
```

Comment:
I still think this is a muddle. The whole notion of the 'center' is kind of confusing here. I'll try to explain:

For the most part, what we're interested in is where all the bits of the detector are relative to one another, so we can assemble a complete image. In the big quadrant-based detectors, we use the centre (where the beam goes through a hole) as the origin, and give the quadrant positions relative to that. This detector doesn't have an obvious central feature, and for assembling an image, it doesn't matter where we put the origin.

But for a second level of analysis, the origin does matter. E.g. if you're doing azimuthal integration, you need a point to integrate around. We've referred to the point where the beam passes through the detector, which works for a detector perpendicular to the beam, but more generally, we want the origin to be where a ray from the sample is perpendicular to the detector plane. PyFAI calls this the PONI, or Point Of Normal Incidence.

I'm not exactly sure how to handle this. Maybe we should have a `AGIPD_500K2GGeometry.default()` class method which constructs it with the origin in the centre of the 8 modules (as an arbitrary choice)?  Do we know anything about how it will be positioned for use?

Reply:
I jsut asked. It will be sitting on a bench at the end of the chamber, perpendicular to the beam direction (similar to SPB). Beam will pass probably above or below the detector. Also learned that the 500K agipd is temporary they will install a full AGIPD at some point.

Do you mean to have 2 methods: 1 `default` from the center (or corner, ..) + 1 from the PONI (how we should call that)?

Reply:
I think that might be the best thing. `default()` with no required arguments, so you can assemble an image without worrying about where the detector is, and something like `from_center` or `from_poni`. I can't decide if 'center' is meaningful enough, or 'PONI' is too much jargon. Maybe it's better to use a word people don't know than one which they might guess an incorrect meaning for.

I also wonder if this second method should invert the numbers, i.e. you give the position of the PONI relative to some part of the detector, rather than giving the position of the sensor relative to the PONI.

Reply:
So, I added a `default` method. with the reference (0, 0) in the bottom-right corner (so pixel indices are always positive), I can change that to the centrer or an other corner if you prefer...
I also changed `from_center` to `from_origin`, is that an OK name for you? (also inverted the numbers: origin position is relative to the detector corner)